runtime/autoload/man.vim

```diff
@@ -0,0 +1,131 @@
+let s:man_tag_depth = 0
+let s:man_sect_arg = ''
+let s:man_find_arg = '-w'
+
+try
+  if !has('win32') && $OSTYPE !~? 'cygwin\|linux' && system('uname -s') =~? 'SunOS' && system('uname -r') =~? '^5'
+    let s:man_sect_arg = '-s'
+    let s:man_find_arg = '-l'
+  endif
+catch /E145:/
+  " Ignore the error in restricted mode
+endtry
+
+" Expects a string like 'access' or 'access(2)'.
+function s:parse_page_and_section(str)
+  let l:page = substitute(a:str, '(*\(\k\+\).*', '\1', '')
+  let l:sect = substitute(a:str, '\(\k\+\)(\([^()]*\)).*', '\2', '')
```

Comment:
There is an extremely bizarre bug (present in at least Vim 7.3 as well) that causes the result of 

    let l:page = substitute(a:str, '(*\(\k\+\).*', '\1', '')

to return `"foo(1)"` if it is called twice in a row on a `filetype=man` page. Haven't narrow it down yet.